dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java

```diff
@@ -223,44 +229,80 @@ private void connect(String lastEventId) {
             if (response.getStatus() == 503) {
                 LOG.fine("SSE endpoint " + target.getUri() + " returns 503");
                 MultivaluedMap<String,Object> headerMap = response.getHeaders();
-                Object retryAfter = headerMap.get(HttpHeaders.RETRY_AFTER);
-                ArrayList retryArray = (ArrayList)retryAfter;
-                if (!retryArray.isEmpty()) {
-                    //There should only be one entry
-                    handleRetry((String)retryArray.get(0));
+                //There should only be one header entry
+                Object retryAfter = headerMap.getFirst(HttpHeaders.RETRY_AFTER);
+                if (retryAfter != null) {
+                    long retryAfterDelay = handleRetry((String)retryAfter);
+                    delay = retryAfterDelay;
+                    if (retryAfterDelay > -1) {
+                        scheduleReconnect(retryAfterDelay, lastEventId);
+                        response.close();
+                        return;
+                    }
+                    
                 }
             }
 
+            int status = response.getStatus();
+            String contentType = response.getHeaderString(HttpHeaders.CONTENT_TYPE);
+            if (status != 200 || !MediaType.SERVER_SENT_EVENTS.equals(contentType)) {
+                if (LOG.isLoggable(Level.FINEST)) {
+                    LOG.log(Level.FINEST, "Received " + status + " Content-Type=" + contentType);
+                }
```

Comment:
Do you want this log message to indicate that something is not right with the content-Type?

Reply:
For this, no - the different WebApplicationException that gets thrown below will also be logged, so it should be apparent why this is an error.